### PR TITLE
[feat] new engine: wikispecies

### LINKED ISSUE #1222: Wikispecies
**Working URL to the engine**
https://species.wikimedia.org

**Why do you want to add this engine?**
Out of all of the Wikimedia projects, Wikimedia Species is one of only two to not appear in any engine category in SearXNG, with the other being Wikimedia Commons.

**Features of this engine**
It has [803,813](https://species.wikimedia.org/wiki/Special:Statistics) articles, covering [Animalia](https://species.wikimedia.org/wiki/Animalia), [Plantae](https://species.wikimedia.org/wiki/Plantae), [Fungi](https://species.wikimedia.org/wiki/Fungi), [Bacteria](https://species.wikimedia.org/wiki/Bacteria), [Archaea](https://species.wikimedia.org/wiki/Archaea), [Protista](https://species.wikimedia.org/wiki/Protista) and all other forms of life.

**How can SearXNG fetch the information from this engine?**
`https://species.wikimedia.org/w/index.php?search=%s` with `%s` being what you want to search.

**Applicable category of this engine**
General, science.

### PR DESCRIPTION
**Pull request by Bnyro**
## What does this PR do?
* adds wikispecies as an engine
* wikispecies is just a standard wikimedia api

## Why is this change important?
* it contains articles about different animals/species and other forms of life.

## How to test this PR locally?
* disable all engines except and enable wikispecies

## Related issues

closes  #1222

### CODE REVIEW

File: searx/engines/mediawiki.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def response(resp):
         )
 
         # append result
-        results.append({'url': url, 'title': result['title'], 'content': ''})
+        results.append({'url': url, 'title': result['title'], 'content': result.get('snippet', '')})
```
Comment (return42) [inline searx/engines/mediawiki.py]: Wow this is *rocket cool*, it (BTW) fixes the content field of wikibooks, wikinews, wikiquote .. and all the other mediawiki engines :+1:

In my review I will strip the HTML tags from the *snippet* .. coming soon.

**Comment by return42** [comment]
@Bnyro I added two commits on top of your branch .. can you please have a look / thanks :+1: 

1. in 5e58ccf8aeb I added some docs and some additional config options for generic wikimedia engines
2. in a96b6f3 I enabled the engine from the wikimedia foundation 

@Bnyro, @unixfox what do you think about the 2nd patch ... is it OK or does it pollute the result list a bit too much?

**Comment by Bnyro** [comment]
That looks awesome in my opinion.
The wikimedia engines provide great results and therefore having them enabled by default is a good decision in my opinion.

**Comment by return42** [comment]
> @Bnyro, @unixfox what do you think about the 2nd patch ... is it OK or does it pollute the result list a bit too much?

After using this setting for a while I'm no longer sure if it was a good decision to activate theses engines by default (a96b6f3a31) / in the *general* category I often get to many results with a ranking on top of the result list. 

By example: when I query for [markdown](https://searx.be/search?q=markdown) I got to many results I didn't was looking for.

Should we better deactivate these engines by default?

---

Or should we rank ([`weight`](https://docs.searxng.org/admin/settings/settings_engine.html)) them lower?

**Comment by Bnyro** [comment]
I think it's better to weight them lower, since they do in fact provide good results.

**Comment by unixfox** [comment]
> > @Bnyro, @unixfox what do you think about the 2nd patch ... is it OK or does it pollute the result list a bit too much?
> 
> After using this setting for a while I'm no longer sure if it was a good decision to activate theses engines by default ([a96b6f3](https://github.com/searxng/searxng/commit/a96b6f3a31f315a77e65016b8be57fd4f486fc9c)) / in the _general_ category I often get to many results with a ranking on top of the result list.
> 
> By example: when I query for [markdown](https://searx.be/search?q=markdown) I got to many results I didn't was looking for.
> 
> Should we better deactivate these engines by default?

I would prefer to deactivate them, I never really got any meaningful results from them. In fact, I have disabled them on searx.be

I think we should be very careful with what engines we add as enabled by default for the General category, some people will not like if we add too many new kind of engines (4 new engines enabled by default in a row is a lot).

**Pull request merged**